modules/control-utility/src/test/java/org/apache/ignite/util/GridCommandHandlerResetLostPartitionTest.java

```diff
@@ -143,17 +148,72 @@ public void testCacheResetLostPartitionsAll() throws Exception {
         stopAllGrids();
         startGrids(3);
 
+        crd = grid(0);
+
         // All data was back.
         assertEquals(CACHE_NAMES.length * CACHE_SIZE, averageSizeAroundAllNodes());
     }
 
+    /**
+     * Tests that resetting lost partitions for the system cache.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testResetLostPartitionsSystemCache() throws Exception {
+        IgniteEx g0 = startGrids(2);
+
+        g0.cluster().state(ACTIVE);
+
+        stopGrid(1);
+
+        // Fill the system cache with some data.
+        // Number of key value pairs should be enough to fill all partitions.
+        IgniteInternalCache<Integer, Integer> sysCache0 = g0.context().cache().utilityCache();
+        int keyNums = sysCache0.configuration().getAffinity().partitions() * 10;
+        for (int i = 0; i < keyNums; i++)
+            sysCache0.put(i, i);
+
+        // Block supply messages for the system cache and start previously stopped node to make it lost partitions.
+        TestRecordingCommunicationSpi spi0 = TestRecordingCommunicationSpi.spi(g0);
+        spi0.blockMessages(TestRecordingCommunicationSpi.blockSupplyMessageForGroup(CU.cacheId(CU.UTILITY_CACHE_NAME)));
+
+        IgniteInternalFuture<?> startFut = GridTestUtils.runAsync(() -> startGrid(1));
+
+        spi0.waitForBlocked();
+
+        stopGrid(0);
+
+        spi0.stopBlock(false);
+
+        startFut.get();
+
+        crd = grid(1);
```

Comment:
This variable comes from a base class and is used by the `afterTest` method. If the coordinator node was restarted this variable should be re-initialized to avoid exceptions after a test.